plugin/src/main/java/org/opensearch/ml/sdkclient/SdkClientModule.java

```diff
@@ -16,48 +16,89 @@
 import org.opensearch.client.opensearch.OpenSearchClient;
 import org.opensearch.client.transport.rest_client.RestClientTransport;
 import org.opensearch.common.inject.AbstractModule;
-import org.opensearch.core.common.Strings;
 import org.opensearch.sdk.SdkClient;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 
+import lombok.extern.log4j.Log4j2;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProviderChain;
+import software.amazon.awssdk.auth.credentials.ContainerCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.InstanceProfileCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+
 /**
  * A module for binding this plugin's desired implementation of {@link SdkClient}.
  */
+@Log4j2
 public class SdkClientModule extends AbstractModule {
 
+    public static final String REMOTE_METADATA_TYPE = "REMOTE_METADATA_TYPE";
     public static final String REMOTE_METADATA_ENDPOINT = "REMOTE_METADATA_ENDPOINT";
     public static final String REGION = "REGION";
+    public static final String REMOTE_OPENSEARCH = "RemoteOpenSearch";
+    public static final String AWS_DYNAMO_DB = "AWSDynamoDB";
 
+    private final String remoteStoreType;
     private final String remoteMetadataEndpoint;
     private final String region; // not using with RestClient
 
     /**
      * Instantiate this module using environment variables
      */
     public SdkClientModule() {
-        this(System.getenv(REMOTE_METADATA_ENDPOINT), System.getenv(REGION));
+        this(System.getenv(REMOTE_METADATA_TYPE), System.getenv(REMOTE_METADATA_ENDPOINT), System.getenv(REGION));
     }
 
     /**
      * Instantiate this module specifying the endpoint and region. Package private for testing.
      * @param remoteMetadataEndpoint The remote endpoint
      * @param region The region
      */
-    SdkClientModule(String remoteMetadataEndpoint, String region) {
+    SdkClientModule(String remoteStoreType, String remoteMetadataEndpoint, String region) {
```

Comment:
should we keep similarity with `REMOTE_METADATA_TYPE` like `remoteMetadataType`?